src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurvey.java

```diff
@@ -0,0 +1,239 @@
+    
+    /** @see #getModuleVersion */
+    public void setModuleVersion(Integer moduleVersion) {
+        this.moduleVersion = moduleVersion;
+    }
+    
+    /** The version of this survey as used to implement optimistic locking. */
+    public Long getVersion() {
+        return version;
+    }
+    
+    /** @see #getVersion */
+    public void setVersion(Long version) {
+        this.version = version;
+    }
+    
+    /** The name of this survey. The name can be changed after creation, 
+     * and should not appear in the UI. */
+    public String getName() {
+        return name;
+    }
+    
+    /** @see #getName */
+    public void setName(String name) {
+        this.name = name;
+    }
+    
+    /** */
+    public String getIdentifier() {
+        return identifier;
+    }
+    
+    /** @see #getIdentifier */
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
+    }
+    
+    /** True if this survey revision has been published, and is accessible to 
+     * users through scheduling. */
+    public boolean isPublished() {
+        return published;
+    }
+    
+    /** @see #isPublished */
+    public void setPublished(boolean published) {
+        this.published = published;
+    }
+    
+    /** Has this survey been logically deleted (an admin can restore it)? */
+    public boolean isDeleted() {
+        return deleted;
+    }
+    
+    /** @see #isDeleted */
+    public void setDeleted(boolean deleted) {
+        this.deleted = deleted;
+    }
+    
+    /**
+     * Gets the upload schema revision that corresponds to this survey. See
+     * {@link org.sagebionetworks.bridge.models.upload.UploadSchema#getRevision} for more details.
+     */
+    public Integer getSchemaRevision() {
+        return schemaRevision;
+    }
+    
+    /** @see #getSchemaRevision */
+    public void setSchemaRevision(Integer schemaRevision) {
+        this.schemaRevision = schemaRevision;
+    }
+    
+    /** An ordered collection of SurveyElement sub-types (in the order they will 
+     * appear in the survey). */
+//    @CollectionTable(name = "SurveyElements", joinColumns = @JoinColumn(name = "surveyGuid", referencedColumnName = "guid"))
+//    @Column(name = "elements")
+//    @OrderColumn(name="order", insertable=true, updatable=true)
+//    @ElementCollection(fetch = FetchType.EAGER)
+    @OneToMany(mappedBy = "surveyGuid", cascade = CascadeType.ALL, orphanRemoval = true, 
+    fetch = FetchType.EAGER, targetEntity=HibernateSurveyElement.class)
+    @OnDelete(action=OnDeleteAction.CASCADE)
+    public List<HibernateSurveyElement> getElements() {
+        return elements;
+    }
+    
+    /** @see #getElements */
+    public void setElements(List<HibernateSurveyElement> elements) {
+        this.elements = elements;
+    }
+    
+    public boolean keysEqual(GuidCreatedOnVersionHolder keys) {
+        return (keys != null && keys.getGuid().equals(guid) && keys.getCreatedOn() == createdOn);
+    }
```

Comment:
This is identical in both DynamoSurvey and HibernateSurvey. Why not move it to the interface as a default implementation?